### PR TITLE
Fix/streamline loading with encryption

### PR DESCRIPTION
**Pull request by ShnitzelKiller**
Loading at start clears saved encryption keys to prevent messing with vanilla loading.
When calling /load, behavior with cloud saving is made consistent with new game loading (download file first, then decrypt or whatever needs to be done).

### CODE REVIEW

**Comment by ShnitzelKiller** [comment]
Also for a given setting of /encrypt, all your saves will have the same salt, and often you'll want to load from save points in the same session. So now, the stored encryption key can be used without needing a password unless the salt is different.

**Comment by thadunge2** [comment]
The problem scenario is as follows:

-You save a new encrypted game and go to bed.
-The next day, you start a new game and /encrypt with the same password.
-Then you try to load your old game.

Even though you're using the same password, you'll have a new randomly-generated salt for that session with your password, and you can't load the password from the save's salt without knowing what the unencrypted password is. So it has to be entered again.

**Comment by ShnitzelKiller** [comment]
Yes, with this PR it will only accelerate loading games that you saved in the same session. Otherwise, you will still be asked to re-enter the password. So it speeds things up in one common use case, but not in another (we would need to keep the password stored in memory, and I don't know if you want to do that). However, as far as I understand, it doesn't hurt; after you load that game, any further saves you make will use the same salt by default, so hopping between them won't require re-entering your password. With the previous behavior, every encrypted load requires a password.

So basically, in the problem scenario, the right way to load a game is to select "load game" at the start, then type in the password when prompted. /encrypt is for when you first want to save your game. For loading there's never a need to type /encrypt, as you will be asked for your password anyway if it is necessary. Unless you explicitly change /encrypt password midway between loading and saving, all saves will have the same salt if they're connected to a single game. So navigating between saves of a single game should be password-free.

**Comment by thadunge2** [comment]
Sorry, maybe I missed something in the code. Does it ask you for a password again if it fails to /load with the current password?

**Comment by ShnitzelKiller** [comment]
yes. Any failure to load will ask for the password and try again with the supplied password.

File: play.py [hunk 1]
```diff
@@ -405,14 +405,14 @@ def play_aidungeon_2():
                     if load_ID.startswith("gs://"):
                         story_manager.cloud = True
                         load_ID = load_ID[5:]
-                    if story_manager.has_encryption():
+                    result = story_manager.load_from_storage(load_ID)
+                    if result is None:
```
Comment (ShnitzelKiller) [inline play.py]: previously only asked for the password if encryption is already enabled; now it will first try to load with the current configuration, including any encryption, and only if that fails ask for a new password. This has the benefit of not requiring the user to type in passwords for loading saves within the same game, and also letting the user enter a password when loading an encrypted file if encryption is currently off.
